### PR TITLE
[AVCe] Update table with default number of active references

### PR DESCRIPTION
**Pull request by romka0017**

### CODE REVIEW

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp [hunk 1]
```diff
@@ -977,20 +977,39 @@ namespace
...
             }
-            else //progressive >= 4K
+            else
             {
-                mfxU16 const DEFAULT_BY_TU[] = { 0, 4, 4, 4, 3, 2, 1, 1 };
-                return DEFAULT_BY_TU[targetUsage];
+                if ((info.Width < 3840 && info.Height < 2160) ||
+                    (info.PicStruct != MFX_PICSTRUCT_PROGRESSIVE))
+                {
+                    mfxU16 const DEFAULT_BY_TU[] = { 0, 8, 6, 3, 3, 3, 1, 1 };
```
Comment (dvrogozh) [inline _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: Where these new values come from? Are they HW or driver limitations for the particular platform and target usage?

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp [hunk 2]
```diff
@@ -977,20 +977,39 @@ namespace
...
+        }
         else if (!isLowPower)
         {
-            if((info.Width < 3840 && info.Height < 2160) ||
-               (info.PicStruct != MFX_PICSTRUCT_PROGRESSIVE))
+            if (platform <= MFX_HW_HSW_ULT)
             {
-                mfxU16 const DEFAULT_BY_TU[] = { 0, 8, 6, 4, 3, 2, 1, 1 };
-                return DEFAULT_BY_TU[targetUsage];
+                if ((info.Width < 3840 && info.Height < 2160) ||
```
Comment (dvrogozh) [inline _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: I have much simpler code change proposal: just drop _all_ code branches with <= HSW and go with the single default setting!

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp [hunk 3]
```diff
@@ -977,20 +977,39 @@ namespace
...
+                    (info.PicStruct != MFX_PICSTRUCT_PROGRESSIVE))
+                {
+                    mfxU16 const DEFAULT_BY_TU[] = { 0, 8, 6, 4, 3, 2, 1, 1 };
```
Comment (mgonchar) [inline _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: code is folded and ```DEFAULT_BY_TU``` defined so many times. I'd suggest to create 2-d array and determine first index, then return the value:
```
return DEFAULT_BY_TU[UsageScenario][targetUsage];
```

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp [hunk 4]
```diff
@@ -1002,13 +1021,20 @@ namespace
 
     mfxU16 GetDefaultMaxNumRefActiveBL0(mfxU32 targetUsage, eMFXHWType platform)
     {
-        if (platform == MFX_HW_IVB || platform == MFX_HW_VLV)
+        if (platform <= MFX_HW_IVB || platform == MFX_HW_VLV)
```
Comment (mgonchar) [inline _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]:
```
   mfxU16 GetDefaultMaxNumRefActiveBL0(mfxU32 targetUsage, eMFXHWType platform)
   {
       if (platform <= MFX_HW_IVB || platform == MFX_HW_VLV)
       {
           return 1;
       }

       constexpr mfxU16 DEFAULT_BY_TU[] = { { 0, 4, 4, 3, 2, 2, 1, 1 }, // platform <= MFX_HW_HSW_ULT
                                            { 0, 4, 4, 2, 2, 2, 1, 1 }
                                          };

       return DEFAULT_BY_TU[platform > MFX_HW_HSW_ULT ? 1 : 0][targetUsage];
   }
```

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp [hunk 5]
```diff
@@ -977,38 +977,47 @@ namespace
...
+        {
             return 1;
-        else if (!isLowPower)
+        }
+
+        constexpr mfxU16 DEFAULT_BY_TU[][8] = {
+        { 0, 8, 6, 4, 3, 2, 1, 1 }, // VME progressive < 4k or interlaced (platform <= MFX_HW_HSW_ULT)
```
Comment (mgonchar) [inline _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: add one tabulation for these lines (4 spaces)

**Pull request merged**